File: app/src/main/java/com/mcal/example/utils/ApktoolHelper.java
package com.mcal.example.utils;

import brut.androlib.ApkBuilder;
import brut.androlib.ApkDecoder;
import brut.androlib.Config;
import brut.androlib.exceptions.AndrolibException;
import brut.common.BrutException;
import brut.directory.DirectoryException;
import brut.directory.ExtFile;
import com.mcal.androlib.utils.Logger;
import org.jetbrains.annotations.NotNull;

import java.io.File;
import java.io.IOException;

public class ApktoolHelper {
    public static void decode(
            @NotNull File apkFile,
            @NotNull File decodeDir,
            @NotNull String frameworkPath,
            @NotNull Logger logger) {
        try {
            Config config = Config.getDefaultConfig();
            config.setDecodeSources(Config.DECODE_SOURCES_SMALI);
            config.setDecodeResources(Config.DECODE_RESOURCES_FULL);
            config.setDecodeAssets(Config.DECODE_ASSETS_NONE);
            config.setDefaultFramework(frameworkPath);
            config.setDecodeAllPackages(true);
            config.setForceDelete(true);
            new ApkDecoder(config, new ExtFile(apkFile), logger).decode(decodeDir);
        } catch (AndrolibException | IOException | DirectoryException e) {
            throw new RuntimeException(e);
        }
    }

    public static void buildProject(
            @NotNull File apkFile,
            @NotNull File decodeDir,
            @NotNull String frameworkPath,
            @NotNull String aaptPath,
            @NotNull String aapt2Path,
            boolean useAapt2,
            @NotNull Logger logger) {
        try {
            logger.info("Aapt version: " + (useAapt2 ? "2" : "1"));
            ExtFile appDecodeDir = new ExtFile(decodeDir);
            Config config = Config.getDefaultConfig();
            config.setUseAapt2(useAapt2);
            config.setAaptPath(aaptPath);
            config.setAapt2Path(aapt2Path);
            config.setDefaultFramework(frameworkPath);
            config.setDecodeAllPackages(false); // aapt and aapt2 not support multi packages
            config.setForceDelete(true);
            new ApkBuilder(config, appDecodeDir, logger).build(apkFile);
        } catch (BrutException e) {
            throw new RuntimeException(e);
        }
    }
}

File: app/src/main/java/com/mcal/example/utils/ApkToolHelper.kt
package com.mcal.example.utils

import brut.androlib.ApkBuilder
import brut.androlib.ApkDecoder
import brut.androlib.Config
import brut.directory.ExtFile
import com.mcal.androlib.utils.Logger
import java.io.File

object ApkToolHelper {
    /**
     * @param toolsDir путь к директории с бинарниками и фреймфорками:
     *      ../toolsDir/1.apk
     *      ../toolsDir/android-framework.jar
     *      ../toolsDir/aapt
     *      ../toolsDir/aapt2
     */
    fun decode(apkPath: File, decodeRootPath: File, toolsDir: String, logger: Logger) {
        try {
            val apkFile = ExtFile(apkPath)
            val config = Config.getDefaultConfig()
            config.setDecodeSources(Config.DECODE_SOURCES_SMALI)
            config.setDecodeResources(Config.DECODE_RESOURCES_FULL)
            config.setDecodeAssets(Config.DECODE_ASSETS_NONE)
            config.setDefaultFramework(toolsDir)
            config.setDecodeAllPackages(true)
            config.setForceDelete(true)
            ApkDecoder(config, apkFile, logger).decode(decodeRootPath)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun buildProject(apkPath: File, decodeRootPath: File, toolsDir: String, logger: Logger) {
        val appDecodeDir = ExtFile(decodeRootPath)
        val config = Config.getDefaultConfig()
        config.setUseAapt2(true)
        config.setAaptPath("$toolsDir/aapt")
        config.setAapt2Path("$toolsDir/aapt2")
        config.setDefaultFramework(toolsDir)
        config.setDecodeAllPackages(true)
        config.setForceDelete(true)
        ApkBuilder(config, appDecodeDir, logger).build(apkPath)
    }
}